cli/src/main/java/com/box/l10n/mojito/cli/command/extraction/ExtractionDiffService.java

```diff
@@ -111,20 +124,72 @@ public ExtractionDiffStatistics computeExtractionDiffStatistics(
     return extractionDiffStatistics;
   }
 
-  public List<AssetExtractionDiff> findAssetExtractionDiffsWithAddedTextUnits(
-      ExtractionDiffPaths extractionDiffPaths) throws MissingExtractionDirectoryException {
+  private static void addPath(
+      String checkerTypeName,
+      Map<String, List<AssetExtractionDiff>> pathsByCheckerTypeName,
+      Optional<AssetExtractionDiff> assetExtractionDiff) {
+    pathsByCheckerTypeName.compute(
+        checkerTypeName,
+        (currentCheckerTypeName, assetExtractionDiffs) -> {
+          if (assetExtractionDiffs == null) {
+            assetExtractionDiffs = new ArrayList<>();
+          }
+          if (assetExtractionDiff.isPresent()) {
+            assetExtractionDiffs.add(assetExtractionDiff.get());
+          }
+          return assetExtractionDiffs;
+        });
+  }
+
+  public Map<String, List<AssetExtractionDiff>> findAssetExtractionDiffsWithAddedTextUnits(
+      ExtractionDiffPaths extractionDiffPaths, String excludedFilesRegex, String allCheckerTypesKey)
+      throws MissingExtractionDirectoryException {
 
     ExtractionPaths baseExtractionPaths = extractionDiffPaths.getBaseExtractorPaths();
     ExtractionPaths currentExtractionPaths = extractionDiffPaths.getCurrentExtractorPaths();
 
     checkExtractionDirectoryExists(baseExtractionPaths);
     checkExtractionDirectoryExists(currentExtractionPaths);
 
-    return extractionDiffPaths
-        .findAllAssetExtractionDiffPaths()
-        .map(path -> objectMapper.readValueUnchecked(path.toFile(), AssetExtractionDiff.class))
-        .filter(assetExtractionDiff -> !assetExtractionDiff.getAddedTextunits().isEmpty())
-        .collect(Collectors.toList());
+    Stream<Path> paths = extractionDiffPaths.findAllAssetExtractionDiffPaths();
```

Comment:
I don't think we should be introducing check logic to the `ExtractionDiffService`, the change here should be isolated to the context comment check itself, if an `AssetExtractorTextUnit` has a usage that matches a provided regex then just mark that text unit as passing the check

Reply:
Done, please review